prime-router/metadata/hl7_mapping/datatypes/HumanName/XPN.yml

```diff
@@ -3,63 +3,70 @@ elements:
   - name: xpn-family-name
     resource: '%resource.family'
     constants:
-      hl7fnField: '%{hl7xpnField}-1'
+      hl7fnField: '%{xpnField}-1'
     schema: ./FN
 
   - name: xpn-given-name
     value: [ '%resource.extension(%`rsext-xpn2-given-name`).value' ]
-    hl7Spec: [ '%{hl7xpnField}-2' ]
+    hl7Spec: [ '%{xpnField}-2' ]
 
   - name: xpn-second-given-name
     value: [ '%resource.extension(%`rsext-xpn3-second-given-name`).value' ]
-    hl7Spec: [ '%{hl7xpnField}-3' ]
+    hl7Spec: [ '%{xpnField}-3' ]
 
   - name: xpn-suffix
     value: [ '%resource.extension(%`rsext-xpn4-suffix`).value' ]
-    hl7Spec: [ '%{hl7xpnField}-4' ]
+    hl7Spec: [ '%{xpnField}-4' ]
 
   - name: xpn-prefix
     value: [ '%resource.prefix' ]
-    hl7Spec: [ '%{hl7xpnField}-5' ]
+    hl7Spec: [ '%{xpnField}-5' ]
 
   - name: xpn-degree
     value: [ '%resource.extension(%`rsext-xpn6-degree`).value' ]
-    hl7Spec: [ '%{hl7xpnField}-6' ]
+    hl7Spec: [ '%{xpnField}-6' ]
 
   - name: xpn-name-type-code
-    value: [ '%resource.extension(%`rsext-xpn7-name-type-code`).value' ]
-    hl7Spec: [ '%{hl7xpnField}-7' ]
+    value: [ '%resource.extension(%`rsext-xpn7-name-type-code`).value', '%resource.use' ]
+    hl7Spec: [ '%{xpnField}-7' ]
+    valueSet:
```

Comment:
Won't this result in not mapping something to XPN.7 if it's not in the valueset?  And why are we not guaranteed to have the extension set?

Reply:
`resource.use` would come from SR. I didn't add the sender transform to create the extension but thought this would cover it. Do you think it should be a sender transform instead? If `resource.use` is not in the valueset then it won't be added to the HL7. The `ca-patient` receiver transform has a default backup for `official`